src/main/java/com/mabl/integration/jenkins/MablSystemException.java

```diff
@@ -0,0 +1,20 @@
+package com.mabl.integration.jenkins;
+
+/**
+ * Non test related failure
+ *
+ */
+class MablSystemException extends Exception {
+
+    public MablSystemException(String message) {
+        super(message);
+    }
+
+    public MablSystemException(String format, Object ... args) {
+        super(String.format(format, args));
+    }
+
+    public MablSystemException(String message, Throwable cause) {
```

Comment:
Just to confirm, Java will choose this method signature overload rather than the vararg variant, during `new MablSystemException("", exception)` based on the more specific match of a an `Exception` to `Throwable` than `Object`?

Reply:
Yes, that's exactly the case.